pyneuroml/swc/ExportNML.py

```diff
@@ -0,0 +1,501 @@
+        :param version: The NeuroML version being used.
+        """
+        print(
+            f"Parsing tree: Point {this_point.id}, Type {this_point.type}, NewCable: {new_cable}, NewCell: {new_cell}"
+        )
+
+        this_type = max(this_point.type, 0)
+        cable_id = -1
+
+        if new_cable:
+            cable_id = self.next_cable_id
+            self.next_cable_id += 1
+            self.cable_ids_vs_indices[this_point.id] = cable_id
+            print(f"New cable created: Cable ID {cable_id}")
+        else:
+            cable_id = self.cable_ids_vs_indices[parent_point.id]
+            self.cable_ids_vs_indices[this_point.id] = cable_id
+            print(f"Using existing cable: Cable ID {cable_id}")
+
+        fract_along_parent_cable = 1
+
+        if this_point.type == SWCNode.SOMA:
+            self.handle_soma(this_point, parent_point, cable_id, version, new_cell)
+        elif this_point.type != SWCNode.SOMA and parent_point.type == SWCNode.SOMA:
+            print("Parent point is on soma! Not creating 'real' segment")
+            self.segment_content.append(
+                "      <!-- Parent point is on soma! Not creating 'real' segment -->"
+            )
+            if (
+                parent_point.id == 0
+                and parent_point.id not in self.point_indices_vs_seg_ids
+            ):
+                fract_along_parent_cable = 0
+        else:
+            print(f"Creating segment for point {this_point.id}")
+            self.create_segment(this_point, parent_point, cable_id, version, new_cable)
+
+        if new_cable:
+            print(f"Handling new cable for point {this_point.id}")
+            self.handle_new_cable(
+                this_point, cable_id, this_type, fract_along_parent_cable, version
+            )
+
+        self.processed_nodes.add(this_point.id)
+
+        num_neighbs_not_done = sum(
+            1 for nbr in this_point.children if nbr.id not in self.processed_nodes
+        )
+        diff_type_any_neighb = any(
+            nbr.type != this_point.type
+            for nbr in this_point.children
+            if nbr.id not in self.processed_nodes
+        )
+
+        for next_point in this_point.children:
+            if next_point.id not in self.processed_nodes:
+                new_cable = diff_type_any_neighb or num_neighbs_not_done > 1
+                if this_point.type == SWCNode.SOMA and next_point.type == SWCNode.SOMA:
+                    new_cable = False
+                    print(f"Continuing soma: {this_point.id} -> {next_point.id}")
+                print(f"Processing child point {next_point.id}, NewCable: {new_cable}")
+                self.parse_tree(this_point, next_point, new_cable, False, version)
+
+    def handle_soma(
+        self,
+        this_point: SWCNode,
+        parent_point: SWCNode,
+        cable_id: int,
+        version: str,
+        new_cell: bool,
+    ):
+        """
+        Handle the soma point in the neural structure.
+
```

Comment:
It'll be good to include an explanation of the different soma cases that we have to handle, with a link to the pdf file. The idea is that someone else reading the code should be able to see the reasoning behind the function (without having to go looking to figure it out themselves). So write a nice long docstring here that explains the multiple cases---it'll also help you check your own understanding of the logic you've used.